### PR TITLE
Update EasyArchiveBag.scala

### PR DESCRIPTION
**Pull request by ekoi**
A solution for issue 36 (https://github.com/DANS-KNAW/easy-archive-bag/issues/6): Wait for SUBMITTED state

### CODE REVIEW

**Comment by ekoi** [comment]
It works, but I am not sure about my solution.

File: src/main/scala/nl/knaw/dans/easy/archivebag/EasyArchiveBag.scala [hunk 1]
```diff
@@ -85,21 +90,64 @@ object EasyArchiveBag {
...
         log.info(s"Deposit created at: $location")
         s"${location.split('/').last}/${s.bagDir.getName}"
+
       case _ =>
         val writer = new StringWriter()
         IOUtils.copy(response.getEntity.getContent, writer, "UTF-8")
         log.error(s"Deposit failed: ${response.getStatusLine}, ${writer.toString}")
         throw new RuntimeException("Deposit failed")
     }
+    log.info(seIri)
```
Comment (janvanmansum) [inline src/main/scala/nl/knaw/dans/easy/archivebag/EasyArchiveBag.scala]: When logging keep in mind:
- Make the log statements informative. How are we supposed to know what the IRI means when we see above statement appear in the log (I mean, without inspecting this code)?
- Choose the `info` level only if appropriate. `info` is what you want the application manager to see. Here `debug` seems more to the point.

File: src/main/scala/nl/knaw/dans/easy/archivebag/EasyArchiveBag.scala [hunk 2]
```diff
@@ -85,21 +90,64 @@ object EasyArchiveBag {
     }
     val deleted = tempFile.delete()
     if(!deleted) log.warn(s"Delete of $tempFile failed")
-    response.getStatusLine.getStatusCode match {
+    val seIri = response.getStatusLine.getStatusCode match {
```
Comment (janvanmansum) [inline src/main/scala/nl/knaw/dans/easy/archivebag/EasyArchiveBag.scala]: I don't think this is actually the SE-IRI from the SWORD specs.

File: src/main/scala/nl/knaw/dans/easy/archivebag/EasyArchiveBag.scala [hunk 3]
```diff
@@ -85,21 +90,64 @@ object EasyArchiveBag {
...
         IOUtils.copy(response.getEntity.getContent, writer, "UTF-8")
         log.error(s"Deposit failed: ${response.getStatusLine}, ${writer.toString}")
         throw new RuntimeException("Deposit failed")
     }
+    log.info(seIri)
+    (seIri, getConfirmationState(seIri, http))
+  }
+
+  def getConfirmationState(locationBagDir : String, http : CloseableHttpClient)(implicit s: Settings) : String = {
+    var state = STATE_FINALIZING
```
Comment (janvanmansum) [inline src/main/scala/nl/knaw/dans/easy/archivebag/EasyArchiveBag.scala]: Don't use vars unless you really have to (which is rarely)!

File: src/main/scala/nl/knaw/dans/easy/archivebag/EasyArchiveBag.scala [hunk 4]
```diff
@@ -85,21 +90,64 @@ object EasyArchiveBag {
...
+    val context = s.storageDepositService.getPath.split('/')(1)
+    val location = locationBagDir.split('/')(0)
+    val url = s"${protocol}://${host}/${context}/statement/${location}"
+
+    //Wait for confirmation that deposit is valid (SUBMITTED or REJECTED)
+    while (state == STATE_FINALIZING) {
+      val content = getResponseContent(url, http)
+      val resp = XML.loadString(content)
+      state = (resp \ "category" \\ "@term").text
+      log.info(s"state: ${state}")
```
Comment (janvanmansum) [inline src/main/scala/nl/knaw/dans/easy/archivebag/EasyArchiveBag.scala]: Be a bit more verbose in your logging statements, something like "Polling storage service for confirmation at $url" or something

File: src/main/scala/nl/knaw/dans/easy/archivebag/EasyArchiveBag.scala [hunk 5]
```diff
@@ -85,21 +90,64 @@ object EasyArchiveBag {
...
+  def getConfirmationState(locationBagDir : String, http : CloseableHttpClient)(implicit s: Settings) : String = {
+    var state = STATE_FINALIZING
+    val protocol = s.storageDepositService.getProtocol
+    val host = s.storageDepositService.getHost
+    val context = s.storageDepositService.getPath.split('/')(1)
+    val location = locationBagDir.split('/')(0)
+    val url = s"${protocol}://${host}/${context}/statement/${location}"
+
+    //Wait for confirmation that deposit is valid (SUBMITTED or REJECTED)
+    while (state == STATE_FINALIZING) {
```
Comment (janvanmansum) [inline src/main/scala/nl/knaw/dans/easy/archivebag/EasyArchiveBag.scala]: What if the storage service gets stuck? We will wait forever.

File: src/main/scala/nl/knaw/dans/easy/archivebag/EasyArchiveBag.scala [hunk 6]
```diff
@@ -85,21 +90,64 @@ object EasyArchiveBag {
...
+    (seIri, getConfirmationState(seIri, http))
+  }
+
+  def getConfirmationState(locationBagDir : String, http : CloseableHttpClient)(implicit s: Settings) : String = {
+    var state = STATE_FINALIZING
+    val protocol = s.storageDepositService.getProtocol
+    val host = s.storageDepositService.getHost
+    val context = s.storageDepositService.getPath.split('/')(1)
+    val location = locationBagDir.split('/')(0)
+    val url = s"${protocol}://${host}/${context}/statement/${location}"
```
Comment (janvanmansum) [inline src/main/scala/nl/knaw/dans/easy/archivebag/EasyArchiveBag.scala]: - It should be possible to create the url with less lines, something like
          
          val url = new URL(s.storageDepositService, s"statement/$location") 
  or something
- What happens if `locationBagDir` is empty?

File: src/main/scala/nl/knaw/dans/easy/archivebag/EasyArchiveBag.scala [hunk 7]
```diff
@@ -85,21 +90,64 @@ object EasyArchiveBag {
...
+
+    //Wait for confirmation that deposit is valid (SUBMITTED or REJECTED)
+    while (state == STATE_FINALIZING) {
+      val content = getResponseContent(url, http)
+      val resp = XML.loadString(content)
+      state = (resp \ "category" \\ "@term").text
```
Comment (janvanmansum) [inline src/main/scala/nl/knaw/dans/easy/archivebag/EasyArchiveBag.scala]: This is incorrect. Simply looking for any category element will not do. The [scheme attribute needs to be set to the correct URI as well](https://github.com/DANS-KNAW/easy-deposit#getting-the-state)

**Pull request closed without merging**